src/wallet/rpcwallet.cpp

```diff
@@ -3376,30 +3376,39 @@ UniValue rescanblockchain(const JSONRPCRequest& request)
             );
     }
 
-    LOCK2(cs_main, pwallet->cs_wallet);
+    WalletRescanReserver reserver(pwallet);
+    if (!reserver.reserve()) {
+        throw JSONRPCError(RPC_WALLET_ERROR, "Wallet is currently rescanning. Abort existing rescan or wait.");
+    }
 
-    CBlockIndex *pindexStart = chainActive.Genesis();
+    CBlockIndex *pindexStart = nullptr;
     CBlockIndex *pindexStop = nullptr;
-    if (!request.params[0].isNull()) {
-        pindexStart = chainActive[request.params[0].get_int()];
-        if (!pindexStart) {
-            throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid start_height");
-        }
-    }
+    CBlockIndex *pChainTip = nullptr;
+    {
```

Comment:
I think you meant to put a LOCK(cs_main) at the top of this scope?

Reply:
Oh. Indeed! Added. Fixed.